cmake/compiler_flags_GNU_Fortran.cmake

```diff
@@ -1,41 +1,40 @@
-# (C) Copyright 2020-2020 NOAA/NWS/NCEP/EMC
+# (C) Copyright 2009-2016 ECMWF.
 #
 # This software is licensed under the terms of the Apache Licence Version 2.0
 # which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation nor
+# does it submit to any jurisdiction.
 
 ####################################################################
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -fbacktrace")
-
-if(CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
```

Comment:
@srherbener  and @PatNichols  do we want to have `-fallow-agrument-mismatch` ?

Reply:
I agree with removing the `-fallow-argument-mismatch` option. I can't think of a case where you would want to allow argument mismatching (of which I think could lead to some quite difficult debugging efforts).